MapleServer2/PacketHandlers/Game/MatchPartyHandler.cs

```diff
@@ -0,0 +1,141 @@
+﻿using MaplePacketLib2.Tools;
+using MapleServer2.Constants;
+using MapleServer2.Enums;
+using MapleServer2.Packets;
+using MapleServer2.Servers.Game;
+using MapleServer2.Tools;
+using MapleServer2.Types;
+using Microsoft.Extensions.Logging;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace MapleServer2.PacketHandlers.Game
+{
+    public class MatchPartyHandler : GamePacketHandler
+    {
+        public override RecvOp OpCode => RecvOp.MATCH_PARTY;
+
+        public MatchPartyHandler(ILogger<MatchPartyHandler> logger) : base(logger) { }
+
+        private enum MatchPartyMode : byte
+        {
+            CreateListing = 0x0,
+            RemoveListing = 0x1,
+            Refresh = 0x2
+        };
+
+        public override void Handle(GameSession session, PacketReader packet)
+        {
+            MatchPartyMode mode = (MatchPartyMode)packet.ReadByte();
+            switch (mode)
+            {
+                case MatchPartyMode.CreateListing:
+                    HandleCreateListing(session, packet);
+                    break;
+                case MatchPartyMode.RemoveListing:
+                    HandleRemoveListing(session, packet);
+                    break;
+                case MatchPartyMode.Refresh:
+                    HandleRefresh(session, packet);
+                    break;
+                default:
+                    IPacketHandler<GameSession>.LogUnknownMode(mode);
+                    break;
+            }
+        }
+
+        public void HandleCreateListing(GameSession session, PacketReader packet)
+        {
+            string partyName = packet.ReadUnicodeString();
+            bool approval = packet.ReadBool();
+            int maxMembers = packet.ReadInt();
+
+            Party party = GameServer.PartyManager.GetPartyByLeader(session.Player);
+
+            if (party == null)
+            {
+                Party newParty = new Party(maxMembers, new List<Player> { session.Player }, partyName, approval);
+                GameServer.PartyManager.AddParty(newParty);
+
+                session.Send(PartyPacket.Create(session.Player));
+                session.Send(PartyPacket.UpdateHitpoints(session.Player));
+
+                session.Player.PartyId = newParty.Id;
```

Comment:
Perhaps if a new Party is created, and it takes a list of members, there should be a loop inside the constructor that automatically sets all member's PartyIds to that new party's Id, so that you don't need to manually do it like here